src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -128,6 +129,36 @@ export class WorkspaceSavedObjectsClientWrapper {
     return false;
   }
 
+  private async validateWorkspacesAndSavedObjectsPermissions(
+    savedObject: Pick<SavedObject, 'id' | 'type' | 'workspaces' | 'permissions'>,
+    request: OpenSearchDashboardsRequest,
+    workspacePermissionModes: WorkspacePermissionMode[],
+    objectPermissionModes: WorkspacePermissionMode[],
+    validateAllWorkspaces = true
+  ) {
+    const { id, type } = savedObject;
+    if (!savedObject.workspaces && !savedObject.permissions) {
+      return true;
+    }
+
+    if (
+      await (validateAllWorkspaces
+        ? this.validateMultiWorkspacesPermissions
+        : this.validateAtLeastOnePermittedWorkspaces
+      ).bind(this)(savedObject.workspaces ?? [], request, workspacePermissionModes)
```

Comment:
What about adding a judge for workspaces length before validation ? It seems that we can return the result instead of doing extra validation if workspaces is an empty array.

Reply:
The `validateMultiWorkspacesPermissions` and `validateAtLeastOnePermittedWorkspaces` will check workspace length in their implementation, so I think we don't need do this check outside.